build-tools/src/main/kotlin/org/jetbrains/kotlin/XcRunRuntimeUtils.kt

```diff
@@ -15,21 +15,17 @@ private fun compareStringsAsVersions(version1: String, version2: String): Int {
     val version2 = version2.split('.').map { it.toInt() }
     val minimalLength = min(version1.size, version2.size)
     for (index in 0 until minimalLength) {
-        if (version1[index] < version2[index]) return -1;
-        if (version1[index] > version2[index]) return 1;
-    }
-    return when {
-        version1.size == version2.size -> 0
-        version1.size < version2.size -> -1
-        else -> 1
+        if (version1[index] < version2[index]) return -1
```

Comment:
Why not use compareTo here as well?

Reply:
Because we want to continue iteration in case of equality.